engine/robot.py

```diff
@@ -330,10 +361,53 @@ def set_phase(self, new_phase):
         with open(CSV_PATH + '/phases.csv', 'a') as fd:
             fd.write(str(self.phase) + '\n')
 
-    def execute_avoid_obstacle(self):
-        # TODO: SET BACK TO ORIGINAL MISSION (TRAVERSE OR RETURN)
+    def track_obstacle(self):
         pass
 
+    def execute_avoid_obstacle(self, dist_to_goal, prev_phase):
+        pass
+
+    def execute_boundary_following(self, min_dist):
+        front_dist = self.rf_ultrasonic.distance()
+        back_dist = self.rb_ultrasonic.distance()
```

Comment:
will there be support for boundary following involving getting the distance on the left side? Please comment which orientation the robot will traverse the obstacle with.